### PR TITLE
feat(trading): use custom toast content for stop order related transactions

### LINKED ISSUE #4450: Customize toast content for stopOrdersSubmission and stopOrdersCancellation transactions
## Story

As a console user I want to see transaction details in toast for stopOrdersCancellation and stopOrdersSubmission transations

### PR DESCRIPTION
**Pull request by bglownia**
# Related issues 🔗

Closes #4450 

# Demo 📺

image

### CODE REVIEW

File: libs/web3/src/lib/use-vega-transaction-toasts.tsx [hunk 1]
```diff
@@ -385,7 +498,7 @@ export const VegaTransactionDetails = ({ tx }: { tx: VegaStoredTxState }) => {
 
   if (isTransferTransaction(tx.body)) {
     const { amount, to, asset } = tx.body.transfer;
-    const transferAsset = assets?.find((a) => a.id === asset);
+    const transferAsset = assets?.[asset];
```
Comment (MadalinaRaicu) [inline libs/web3/src/lib/use-vega-transaction-toasts.tsx:501]: I'm a bit unclear how this works

**Review by jeremyletang — APPROVED** [review]

**Review by guoguojin — APPROVED** [review]

File: libs/web3/src/lib/use-vega-transaction-toasts.spec.tsx [hunk 2]
```diff
@@ -59,20 +60,64 @@ jest.mock('@vegaprotocol/orders', () => {
...
+            size: '10',
+            remaining: '10',
+            timeInForce: 'TIME_IN_FORCE_FOK',
+            type: 'TYPE_MARKET',
+            price: '1234',
+            createdAt: new Date(),
+            status: 'STATUS_ACTIVE',
+            market: { id: 'market-1' },
+          },
+        } as OrderByIdQuery;
```
Comment (mattrussell36) [inline libs/web3/src/lib/use-vega-transaction-toasts.spec.tsx:76]: Would be better to remove the type cast and use the correct enums

File: libs/web3/src/lib/use-vega-transaction-toasts.spec.tsx [hunk 3]
```diff
@@ -59,20 +60,64 @@ jest.mock('@vegaprotocol/orders', () => {
...
-              market: { id: 'market-1' },
+              reference: '',
+              peggedOrder: null,
+              postOnly: false,
+              reduceOnly: true,
+              __typename: 'OrderSubmission',
             },
+            __typename: 'StopOrder',
           },
+        } as StopOrderByIdQuery;
```
Comment (mattrussell36) [inline libs/web3/src/lib/use-vega-transaction-toasts.spec.tsx:118]: Same as above. Using the correct enums should let you not need the type cast

File: libs/web3/src/lib/use-vega-transaction-toasts.tsx [hunk 4]
```diff
@@ -38,19 +38,20 @@ import {
...
   getRejectionReason,
   useOrderByIdQuery,
+  useStopOrderByIdQuery,
 } from '@vegaprotocol/orders';
-import { useMarketList } from '@vegaprotocol/markets';
+import { useMarketsMapProvider } from '@vegaprotocol/markets';
 import type { Side } from '@vegaprotocol/types';
 import { OrderStatusMapping } from '@vegaprotocol/types';
-import { Size } from '@vegaprotocol/datagrid';
+import { Size, formatTrigger } from '@vegaprotocol/datagrid';
```
Comment (mattrussell36) [inline libs/web3/src/lib/use-vega-transaction-toasts.tsx:54]: I see that size is already imported here.. but for me the web3 lib and more specifically the toast component should not be depending on components from the datagrid lib.

I'd move the formatTrigger function to utils

**Review by pscott31 — APPROVED** [review]

File: libs/web3/src/lib/use-vega-transaction-toasts.spec.tsx (hunk 2 above)
Comment (bglownia) [reply libs/web3/src/lib/use-vega-transaction-toasts.spec.tsx:76]: The module factory of `jest.mock()` is not allowed to reference any out-of-scope variables

File: libs/web3/src/lib/use-vega-transaction-toasts.spec.tsx (hunk 3 above)
Comment (bglownia) [reply libs/web3/src/lib/use-vega-transaction-toasts.spec.tsx:118]: The module factory of `jest.mock()` is not allowed to reference any out-of-scope variables

**Review by mattrussell36 — APPROVED** [review]

**Pull request merged**